### PR TITLE
Add datashuttle blog

### PR DESCRIPTION
**Pull request by JoeZiminski**
This blog is the companion piece to #96. It's aims are to introduce datashuttle as a tool, giving an overview of the benefits of using it and its features.

### CODE REVIEW

**Comment by adamltyson** [comment]
> I think the only way around this is to use the light-mode for everything because it looks the least-worse.

Yeah I would just use the light mode image here. It looks nicer to have an image with a white background in dark mode, than an invisible image in lightmode. Also, I suspect most people will view the website in light mode? I use dark mode for most things, but not stuff like this.

File: docs/source/blog/datashuttle.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+:width: 650px
+```
+```{image} /_static/blog_images/datashuttle/datashuttle-overview-light.png
+:align: center
+:class: only-light
+:width: 650px
+```
+<br>
+
+Maintaining a well-organised neuroscience project is hard.
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:28]:
```suggestion
**Maintaining a well-organised neuroscience project is hard.**
```

File: docs/source/blog/datashuttle.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
+---
+blogpost: true
+date: 25 April 2024
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:3]: BTW these dates can be set in the future if needed.

File: docs/source/blog/datashuttle.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+An immediate benefit of a widely-used standard is that the entire community
+can contribute to shared tools for project management.
+This means individual researchers don't waste time
+writing data-management code. 
+
+In this blog post we introduce 
+[**datashuttle**](https://datashuttle.neuroinformatics.dev/)—a 
+tool for the automated creation, 
+validation and transfer of projects organised to 
+the **NeuroBlueprint** standard. **datashutte** aims to
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:58]:
```suggestion
the **NeuroBlueprint** standard. **datashuttle** aims to
```

File: docs/source/blog/datashuttle.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+drop into existing acquisition pipelines, reducing errors
+associated with manual folder creation and removing the need
+to write your down data-management code.
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:61]: What does this mean?

File: docs/source/blog/datashuttle.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+Below we give a whistlestop tour of **datashuttles** and it's key
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:63]:
```suggestion
Below we give a whistlestop tour of **datashuttle** and it's key
```

File: docs/source/blog/datashuttle.md [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+features. More information is available at the 
+[datashutte](https://datashuttle.neuroinformatics.dev/)
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:65]:
```suggestion
[datashuttle](https://datashuttle.neuroinformatics.dev/) 
```

File: docs/source/blog/datashuttle.md [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+:width: 400px
+```
+<br>
+
+Imagine that you are starting a new experiment and have the first
+acquisition session, or behaviour (`behav`) 
+and electrophysiological (`ephys`) data. 
+
+The first thing typically done prior to acquiring data is to
+create the folders that the data will be stored.
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:87]:
```suggestion
create the folders that the data will be stored in. 
```

File: docs/source/blog/datashuttle.md [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+**datashuttle** transfers the newly acquired data 
+to a central storage machine to be backed up.
+
+Later on in the experiment, you may want to transfer only a subset
+of data from the central machine to an analysis machine. You may 
+want to pilot a behavioural data analysis, and grab
+for example only the behavioural 'test' session for
+the first 5 subjects. **datashuttle** allows flexible custom transfers
+easily, meaning you don't have to drag and drop these data manually or
+write a custom script.
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:103]: Would it be worth rephrasing this as:

When organising data manually you would
* A (which can cause error X)
* B (which can cause error Y)
* C (which can cause error Z)

With datashuttle you can:
*
*

File: docs/source/blog/datashuttle.md [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+
+And that really is all there is to **datashuttle**, a tool to drop into acquisition
+workflows to make folder creation and transfer more convenient and ensure standardisation.
+Below we will give a brief tour of these key **datashuttle** features.
+
+## Creating folders with live-validation
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:109]: Maybe before this give very brief details about installation, and how the GUI runs? Just to illustrate that it's pretty simple and runs anywhere?

File: docs/source/blog/datashuttle.md [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+```{image} /_static/blog_images/datashuttle/create-folders-example-light.png
+:align: center
+:class: only-light
+:width: 650px
+```
+<br>
+
+There are a number of convenience shortcuts to ensure you only need to enter 
+your custom information. For example, the convenience tags 
+(`@DATE@`, `@TIME@`, `@DATETIME@`) will fill the created folder
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:129]: If a naive reader doesn't know what a "convenience tag" is, they may not see them as being very convenient!

File: docs/source/blog/datashuttle.md (hunk 9 above)
Comment (adamltyson) [reply docs/source/blog/datashuttle.md:109]: Also state early that there's a GUI and Python API that can be used interoperably?

**Review by adamltyson — COMMENTED** [review]
Looks nice @JoeZiminski, I've added a few comments.

File: docs/source/blog/datashuttle.md [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+Although everyone can appreciate the benefits of a tidy project
+folder, the practicalities of running an experiment often gets 
+in the way. Folder organisation 
+is low on the list of priorities when it
+comes to acquisition sessions spent managing complex systems 
+and experimental animals.
+
+However, the cost of small mistakes during data acquisition can be high.
+One misplaced character may mean sessions are missed by analysis 
+scripts or subject identifications duplicated.
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:39]: I feel there's something syntactically wrong wit this sentence, some alternatives:

1. One misplaced character may mean sessions are missed by analysis 
scripts or subject IDs are duplicated.

2. One misplaced character may lead to subjects with duplicate IDs or to sessions being skipped by analysis scripts.

File: docs/source/blog/datashuttle.md [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+The best protection against such errors is automating the process
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:40]: This sentence is a bit long-winded, because of the dashes etc. Consider rewording or breaking up.

File: docs/source/blog/datashuttle.md [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
+In our previous blog, we highlighted the benefits of data standardisation 
+for systems neuroscience, introducing the 
+[NeuroBlueprint](https://neuroblueprint.neuroinformatics.dev/) 
+specification. 
+An immediate benefit of a widely-used standard is that the entire community
+can contribute to shared tools for project management.
+This means individual researchers don't waste time
+writing data-management code. 
+
+In this blog post we introduce
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:54]: Earlier you say "In our previous blog", while here you say "In this blog post". I think it should be "blog post" in both cases.

File: docs/source/blog/datashuttle.md (hunk 5 above)
Comment (niksirbi) [reply docs/source/blog/datashuttle.md]: It's the first time a hear the term "whistlestop". Maybe replace with "quick" or something else?

File: docs/source/blog/datashuttle.md [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+[NeuroBlueprint](https://neuroblueprint.neuroinformatics.dev/) 
+specification. 
+An immediate benefit of a widely-used standard is that the entire community
+can contribute to shared tools for project management.
+This means individual researchers don't waste time
+writing data-management code. 
+
+In this blog post we introduce 
+[**datashuttle**](https://datashuttle.neuroinformatics.dev/)—a 
+tool for the automated creation,
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:56]: I like this tagline.

File: docs/source/blog/datashuttle.md [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+:width: 400px
+```
+```{image} /_static/blog_images/datashuttle/tutorial-1-example-file-tree-light.png
+:align: center
+:class: only-light
+:width: 400px
+```
+<br>
+
+Imagine that you are starting a new experiment and have the first
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:82]: "during which you acquire both behavioural (`behav`) and electrophysiological (`ephys`) data" ?

File: docs/source/blog/datashuttle.md [hunk 16]
```diff
@@ -0,0 +1,220 @@
...
+acquisition session, or behaviour (`behav`) 
+and electrophysiological (`ephys`) data. 
+
+The first thing typically done prior to acquiring data is to
+create the folders that the data will be stored. 
+**datashuttle** can be used to quickly create the standardised
+project folders in which to store the acquired data. 
+Using **datashuttle** as apposed to manual creation or using custom
+scripts is that it ensures no formatting errors creep in.
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:91]:
```suggestion
scripts ensures that no formatting errors creep in.
```

File: docs/source/blog/datashuttle.md [hunk 17]
```diff
@@ -0,0 +1,220 @@
...
+
+At the end of the session once data is acquired,
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:93]:
```suggestion
At the end of the data acquisition session,
```

File: docs/source/blog/datashuttle.md [hunk 18]
```diff
@@ -0,0 +1,220 @@
...
+project folders in which to store the acquired data. 
+Using **datashuttle** as apposed to manual creation or using custom
+scripts is that it ensures no formatting errors creep in.
+
+At the end of the session once data is acquired, 
+**datashuttle** transfers the newly acquired data 
+to a central storage machine to be backed up.
+
+Later on in the experiment, you may want to transfer only a subset
+of data from the central machine to an analysis machine. You may
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:98]: "to a computes used for data analysis"? 
"analysis machine" may not be immediately obvious without explanation.

File: docs/source/blog/datashuttle.md (hunk 15 above)
Comment (niksirbi) [reply docs/source/blog/datashuttle.md:82]: also, this sentence should probably go before the image showing the data.

File: docs/source/blog/datashuttle.md [hunk 19]
```diff
@@ -0,0 +1,220 @@
...
+At the end of the session once data is acquired, 
+**datashuttle** transfers the newly acquired data 
+to a central storage machine to be backed up.
+
+Later on in the experiment, you may want to transfer only a subset
+of data from the central machine to an analysis machine. You may 
+want to pilot a behavioural data analysis, and grab
+for example only the behavioural 'test' session for
+the first 5 subjects. **datashuttle** allows flexible custom transfers
+easily, meaning you don't have to drag and drop these data manually or
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:102]: "datashuttle allows for easy and flexible data transfers"?
Saying both "flexible" and "custom" feels redundant.

File: docs/source/blog/datashuttle.md [hunk 20]
```diff
@@ -0,0 +1,220 @@
...
+scripts is that it ensures no formatting errors creep in.
+
+At the end of the session once data is acquired, 
+**datashuttle** transfers the newly acquired data 
+to a central storage machine to be backed up.
+
+Later on in the experiment, you may want to transfer only a subset
+of data from the central machine to an analysis machine. You may 
+want to pilot a behavioural data analysis, and grab
+for example only the behavioural 'test' session for
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:100]: I think we can skip mentioning 'test' sessions to avoid confusion.
Fetching only behavioural data from the first 5 subjects is a clear enough example.

File: docs/source/blog/datashuttle.md [hunk 21]
```diff
@@ -0,0 +1,220 @@
...
+easily, meaning you don't have to drag and drop these data manually or
+write a custom script.
+
+And that really is all there is to **datashuttle**, a tool to drop into acquisition
+workflows to make folder creation and transfer more convenient and ensure standardisation.
+Below we will give a brief tour of these key **datashuttle** features.
+
+## Creating folders with live-validation
+
+To create folders manually through the graphical interface with **datashuttle**,
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:111]: Slightly grammatically awkward.
I'd rephrase as "Creating folders through **datashuttle**'s graphical interface is as simple as..."

File: docs/source/blog/datashuttle.md [hunk 22]
```diff
@@ -0,0 +1,220 @@
...
+
+We have given a brief tour of **datashuttle**'s key features,
+but full details on getting started can be found at our 
+[website](https://datashuttle.neuroinformatics.dev/) and
+[getting started tutorial](https://datashuttle.neuroinformatics.dev/pages/tutorials/getting_started.html).
+
+We are very keen to get feedback on **datashuttle**. 
+Standardisation is incredibly useful, but it should not come at the 
+expense of convenience. **datashuttle** aims to make managing your project easier than 
+it currently is—if it is not, we want to hear how it can be improved.
```
Comment (niksirbi) [inline docs/source/blog/datashuttle.md:214]: Nice!

**Review by niksirbi — COMMENTED** [review]
Any way you can put your name first @JoeZiminski ? You did most of the work on datashuttle and on this blogpost.

File: docs/source/blog/datashuttle.md (hunk 8 above)
Comment (JoeZiminski) [reply docs/source/blog/datashuttle.md:103]: I like the idea of laying it out like that. One issue that is a general awkwardness of this blogpost is it is trying to cater to both those who do data management manually as well as automated. The only worry about emphasising the (many) potential errors caused by manual too much is people who are using scripts think 'well, that's not a problem for me' and ignore it.

File: docs/source/blog/datashuttle.md (hunk 4 above)
Comment (JoeZiminski) [reply docs/source/blog/datashuttle.md:61]: Hmm I think this is an example of the below `awkwardness of this blogpost is it is trying to cater to both those who do data management manually as well as automated.`

It is basically saying three things at once:
- datashuttle should just drop into your existing pipeline at the stage of creating folders, and transferring them at the end of the session
- if you are creating your folders manually, it will help reduce typographical errors
- if you are writing your down data manangement code, it means you no longer have to.

Comment (adamltyson) [reply docs/source/blog/datashuttle.md:61]: This sentence makes sense, I think I must have commented on the wrong line!

File: docs/source/blog/datashuttle.md (hunk 12 above)
Comment (JoeZiminski) [reply docs/source/blog/datashuttle.md:40]: How about:

```
The best protection against such errors is automating the process
through acquisition scripts—resulting in hours spent writing 
data-management code entirely unrelated to the central research goals.
```
I am also not sure whether so say `your` research goals rather than `the` to make it more personal. But it might be too much.

File: docs/source/blog/datashuttle.md (hunk 8 above)
Comment (adamltyson) [reply docs/source/blog/datashuttle.md:103]: Oh yeah, I hadn't thought about our two "target markets!

File: docs/source/blog/datashuttle.md (hunk 4 above)
Comment (niksirbi) [reply docs/source/blog/datashuttle.md:55]: I like how you have made the point here in these 3 bullet points. Maybe that's how we should argue it?

File: docs/source/blog/datashuttle.md (hunk 12 above)
Comment (niksirbi) [reply docs/source/blog/datashuttle.md:34]: I'd use "your" in this case, but more consistency throughout is more important.

File: docs/source/blog/datashuttle.md (hunk 11 above)
Comment (JoeZiminski) [reply docs/source/blog/datashuttle.md:30]: How about `subject identifiers`? I want to avoid ID if possible just to avoid abbreviations.

File: docs/source/blog/datashuttle.md (hunk 4 above)
Comment (JoeZiminski) [reply docs/source/blog/datashuttle.md:52]: >This sentence makes sense, I think I must have commented on the wrong line!

Oh I think it must be the "**down** data-management code." instead of own 😅

File: docs/source/blog/datashuttle.md (hunk 9 above)
Comment (JoeZiminski) [reply docs/source/blog/datashuttle.md:100]: > Maybe before this give very brief details about installation, and how the GUI runs? Just to illustrate that it's pretty simple and runs anywhere?

I'm in two minds about this, it is nice to give a quick overview of getting setup. The issue is there is a function to run `make_config_file` which requires some understanding of local vs. central path concept, which it might be a bit too much to get into here. I don't want to give the impression you can just install and run `datashuttle launch` then start creating files as it might be confusing when people actually go to use it. At the same time I dont really want to discuss the setup as it is TMI🤔 so currently just ignore entirely.

File: docs/source/blog/datashuttle.md (hunk 4 above)
Comment (JoeZiminski) [reply docs/source/blog/datashuttle.md:52]: how about:

```
And that really is all there is to **datashuttle**, which aims to, 
drop into your existing acquisition pipelines whether they be manual
or automated:
- the **graphical interface** can be used instead of manual folder creation, 
reducing the risk of typographical errors
- alternatively, the **Python API** can be used directly in automated acquisition 
pipelines, removing the need to write you own data-management code

Below we will give an overview of **datashttle**'s key folder creation
and transfer features.
```

(also introducing the two ways of running it here).

Comment (adamltyson) [reply docs/source/blog/datashuttle.md:50]: Good to introduce the two ways of running it, but that first sentence is kinda weird (and has dodgy grammar). 

How about:

>*datashuttle**, aims to drop into your existing acquisition pipelines whether they be manual
or automated, with two ways to use it:
>- The **graphical user interface** can be used instead of manual folder creation, 
reducing the risk of typographical errors
>- The **Python API** can be used directly in automated acquisition 
pipelines, removing the need to write you own data-management code

>Below we will give an overview of **datashttle**'s key folder creation
and transfer features.

File: docs/source/blog/datashuttle.md (hunk 9 above)
Comment (adamltyson) [reply docs/source/blog/datashuttle.md:106]: Maybe describe it at a high level? Mention install via conda or PyPi, and runs on any system (windows, Linux, macOS), including via remote terminals?

Comment (adamltyson) [reply docs/source/blog/datashuttle.md:106]: Just to get ahead of anyone thinking "The GUI looks nice, but I usually work remotely via ssh, so this won't work me", or something.

File: docs/source/blog/datashuttle.md (hunk 11 above)
Comment (niksirbi) [reply docs/source/blog/datashuttle.md]: identifiers is fine.

File: docs/source/blog/datashuttle.md (hunk 4 above)
Comment (JoeZiminski) [reply docs/source/blog/datashuttle.md:50]: Great thanks!

File: docs/source/blog/datashuttle.md (hunk 9 above)
Comment (JoeZiminski) [reply docs/source/blog/datashuttle.md:110]: I added the below to the start of the 'How datashuttle is used in an experiment', let me know how it reads / what you think of its placement

**datashuttle** runs on Windows, macOS or Linux and is
[easy to install](https://datashuttle.neuroinformatics.dev/pages/how_tos/install.html)
through 
[Conda Forge](https://conda-forge.org/)
or 
[PyPi](https://pypi.org/). **datashuttle** can be used
from within Python code (using the *Python API*) or through a graphical 
interface that works in any system terminal.

**Comment by JoeZiminski** [comment]
Hey finished with some final tidy ups and tweaks. Could tweak it forever! Would be great if someone can have a final read-through then happy for it to be merged next week.

File: docs/source/blog/datashuttle.md (hunk 9 above)
Comment (adamltyson) [reply docs/source/blog/datashuttle.md:111]: LGTM

File: docs/source/blog/datashuttle.md [hunk 23]
```diff
@@ -0,0 +1,242 @@
...
+- The **Python API** can be used directly in automated acquisition
+pipelines, removing the need to write you own data-management code
+
+Below we will give an overview of **datashuttle**'s key folder creation
+and transfer features.
+
+## Creating folders with live-validation
+
+Creating folders through **datashuttle**'s graphical interface is as simple as
+entering the subject and session names, selecting the datatype and clicking 'create':
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:114]: For consistency with below.
```suggestion
entering the subject and session names, selecting the datatype and clicking `Create Folders`
```

File: docs/source/blog/datashuttle.md [hunk 24]
```diff
@@ -0,0 +1,242 @@
...
+
+```python
+from datashuttle import DataShuttle             # import datashuttle
+
+project = DataShuttle("my_first_project")       # initialise the project
+
+created_folder_paths = project.create_folders(  # create the folders
+    "sub-001", "ses-001_@DATE@", ["behav", "funcimg"]
+)
+```
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:164]: I don't think the comments add much, and we don't have them below.
```suggestion
```python
from datashuttle import DataShuttle

project = DataShuttle("my_first_project")

created_folder_paths = project.create_folders(
    "sub-001", "ses-001_@DATE@", ["behav", "funcimg"]
)
```
```

File: docs/source/blog/datashuttle.md [hunk 25]
```diff
@@ -0,0 +1,242 @@
...
+:align: center
+:class: only-light
+:width: 650px
+```
+<br>
+
+## Getting started with **datashuttle**
+
+We have given a brief tour of **datashuttle**'s key features,
+but full details on getting started can be found on our
```
Comment (adamltyson) [inline docs/source/blog/datashuttle.md:230]:
```suggestion
but full details on getting started can be found on the 
```

File: docs/source/blog/datashuttle.md [hunk 26]
```diff
@@ -0,0 +1,242 @@
...
+```
+```{image} /_static/blog_images/datashuttle/validation-bad-light.png
+:align: center
+:class: only-light
+:width: 500px
+```
+<br>
+
+There are a number of shortcuts to ensure only
+information relevant to your specific project must be typed.
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply docs/source/blog/datashuttle.md:145]: I also coudn't parse this sentence. I'll rewrite it as:
"There are a number of shortcuts to reduce the amount of manual typing."

**Comment by niksirbi** [comment]
Hey @adamltyson, I did a final pass on this post, caught a few more typos and formatting errors, and very slightly rephrased some sentences (ensuring, for example, that data are consistently plural).

I also cross-linked the two blogposts in their introductions. I'm merging this now and we can start sharing the blog posts tomorrow.
